nsxt/resource_nsxt_policy_role_binding.go

```diff
@@ -71,38 +69,44 @@ func resourceNsxtPolicyUserManagementRoleBinding() *schema.Resource {
 				Optional:     true,
 				ValidateFunc: validation.StringInSlice(roleBindingIdentitySourceTypes, false),
 			},
-			"roles_for_path": {
-				Type:        schema.TypeList,
-				Description: "List of roles that are associated with the user, limiting them to a path",
-				Required:    true,
-				Elem: &schema.Resource{
-					Schema: map[string]*schema.Schema{
-						"path": {
-							Type:        schema.TypeString,
-							Description: "Path of the entity in parent hierarchy.",
-							Required:    true,
-						},
-						"role": {
-							Type:        schema.TypeList,
-							Description: "Applicable roles",
-							Required:    true,
-							Elem: &schema.Resource{
-								Schema: map[string]*schema.Schema{
-									"role": {
-										Type:        schema.TypeString,
-										Description: "Short identifier for the role",
-										Required:    true,
-										ValidateFunc: validation.StringMatch(
-											regexp.MustCompile(
-												`^[_a-z0-9-]+$`),
-											"Must be a valid role identifier matching: ^[_a-z0-9-]+$"),
-									},
-									"role_display_name": {
-										Type:        schema.TypeString,
-										Description: "Display name for role",
-										Computed:    true,
-									},
-								},
+			"roles_for_path": getRolesForPathSchema(false),
+		},
+	}
+}
+
+// getRolesForPathSchema return schema for RolesForPath, which is shared between role bindings and PI
+func getRolesForPathSchema(forceNew bool) *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "List of roles that are associated with the user, limiting them to a path",
+		Required:    true,
+		ForceNew:    forceNew,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"path": {
+					Type:        schema.TypeString,
+					Description: "Path of the entity in parent hierarchy.",
+					Required:    true,
+				},
+				"role": {
+					Type:        schema.TypeList,
```

Comment:
I wonder if list of sub-clauses here is justified. Does role_display_name convey useful information?
If not, seems more user friendly to just offer a list of string roles? Unless we expect this object to grow in API

Reply:
fixed